tests/KdybyTests/Validator/Extension.phpt

```diff
@@ -63,6 +64,24 @@ class ExtensionTest extends Tester\TestCase
 		Tester\Assert::same(0, count($violations));
 	}
 
+
+
+	public function testConstraintValidatorFactory()
+	{
+		$container = $this->createContainer();
+
+		$factory = $container->getByType('Symfony\Component\Validator\ConstraintValidatorFactoryInterface');
+
+		// One of the default validators.
+		Tester\Assert::type('Symfony\Component\Validator\Constraints\BlankValidator', $factory->getInstance(new \Symfony\Component\Validator\Constraints\Blank()));
+
+		// ExpressionValidator requires a hack.
+		Tester\Assert::type('Symfony\Component\Validator\Constraints\ExpressionValidator', $factory->getInstance(new \Symfony\Component\Validator\Constraints\Expression(array('expression' => ''))));
+
+		// Custom validator.
+		Tester\Assert::type('KdybyTests\ValidatorMock\CustomConstraintValidator', $factory->getInstance(new \KdybyTests\ValidatorMock\CustomConstraint()));
```

Comment:
`KdybyTests\ValidatorMock\CustomConstraint::validatedBy()` returns 'KdybyTests\ValidatorMock\CustomConstraintValidator'. ConstraintValidatorFactory knows which service is that because of the tag (see nette-reset.neon) and returns that service.